app/controllers/return_controller.js

```diff
@@ -0,0 +1,40 @@
+var api = require(__dirname + '/../utils/api.js');
+var response = require(__dirname + '/../utils/response.js').response;
+
+var Client = require('node-rest-client').Client;
+var client = new Client();
+
+module.exports.bindRoutesTo = (app) => {
+  var RETURN_PATH = "/return/";
+  var PAY_API_PAYMENTS_PATH = '/v1/payments/';
+  
+  app.get(RETURN_PATH + ':paymentReference', (req, res) => {
+    var paymentReference = req.params.paymentReference;
+    var paymentId = req.state[paymentReference].pid;
```

Comment:
While not strictly necessary for a sample integration app, it would be good to check both paymentReference and paymentId are defined.